src/lib/core/evio.c

```diff
@@ -404,19 +404,32 @@ evio_service_bind(struct evio_service *service, const char *uri)
 /** It's safe to stop a service which is not started yet. */
 void
 evio_service_stop(struct evio_service *service)
+{
+	int service_fd = service->ev.fd;
+	evio_service_detach(service);
+	if (service_fd >= 0) {
```

Comment:
You could significantly reduce the indentation level by doing early returns. For instance, when `service_fd < 0`, then if  `service->addr.sa_family != AF_UNIX`. Would save 16 symbols for unlink and the error message, which is a lot. Up to you though.

Reply:
Fixed